src/connectors/aws_flow_log.py

```diff
@@ -0,0 +1,224 @@
+
+    db.create_table(
+        name=landing_table,
+        cols=LANDING_TABLE_COLUMNS,
+        comment=comment
+    )
+
+    db.create_table_from_csv(
+        name='data.network_protocol_mapping',
+        columns=PROTOCOL_MAPPING_TABLE_COLUMNS,
+        file_path=NETWORK_PROTOCOL_PATH,
+        file_format=PROTOCOL_FILE_FORMAT,
+        ifnotexists=True,
+    )
+
+    stage_props = db.fetch_props(
+        f'DESC STAGE {stage}',
+        filter=('AWS_EXTERNAL_ID', 'SNOWFLAKE_IAM_USER')
+    )
+
+    prefix = prefix.rstrip('/')
+
+    return {
+        'newStage': 'created',
+        'newMessage': CONNECT_RESPONSE_MESSAGE.format(
+            role=role,
+            role_trust_relationship=dumps({
+                "Version": "2012-10-17",
+                "Statement": [
+                    {
+                        "Effect": "Allow",
+                        "Principal": {
+                            "AWS": stage_props['SNOWFLAKE_IAM_USER']
+                        },
+                        "Action": "sts:AssumeRole",
+                        "Condition": {
+                            "StringEquals": {
+                                "sts:ExternalId": stage_props['AWS_EXTERNAL_ID']
+                            }
+                        }
+                    }
+                ]
+            }, indent=4),
+            role_policy=dumps({
+                "Version": "2012-10-17",
+                "Statement": [
+                    {
+                        "Effect": "Allow",
+                        "Action": [
+                            "s3:GetObject",
+                            "s3:GetObjectVersion",
+                        ],
+                        "Resource": f"arn:aws:s3:::{bucket}/{prefix}/*"
+                    },
+                    {
+                        "Effect": "Allow",
+                        "Action": "s3:ListBucket",
+                        "Resource": f"arn:aws:s3:::{bucket}",
+                        "Condition": {
+                            "StringLike": {
+                                "s3:prefix": [
+                                    f"{prefix}/*"
+                                ]
+                            }
+                        }
+                    }
+                ]
+            }, indent=4),
+        )
+    }
+
+
+def finalize(connection_name):
+    base_name = f'aws_vpc_flow_log_{connection_name}'
+    pipe = f'data.{base_name}_pipe'
+
+    # IAM change takes 5-15 seconds to take effect
+    sleep(5)
+    db.retry(
+        lambda: db.create_pipe(
+            name=pipe,
+            sql=(
+                f'COPY INTO data.{base_name}_connection '
+                f'FROM (SELECT $1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13, $14 '
+                f'FROM @data.{base_name}_stage/)'
+            ),
+            replace=True,
+            autoingest=True,
+        ),
+        n=10,
+        sleep_seconds_btw_retry=1
+    )
+
+    db.execute(f'ALTER PIPE {pipe} REFRESH')
```

Comment:
this seems like it will work well if they just turned on flow logs, but if they have had it running for awhile, this will block for 60 minutes, before swallowing the error?

Reply:
Do we have a way to resolve this issue? Config faces the same dilemma.

Reply:
Not sure what's available, but if there is a way to async enqueue this and continue down the function, that would probably be ideal. If dropped SQS is still a problem, we could probably solve both by having a stored procedure that maintains a set of TASKs that refresh a subset of the PIPE, maybe one per account as we had elsewhere?

The way that would work for sure is to make this an hourly task, but that seems like it may be worse than having anything at all, as it would leave our biggest partners in a place where they have a partial refresh running every hour.

Reply:
How about just skipping it and telling users to run it manually?

Reply:
That works for me; the only reason you need to refresh the pipe is to pick up old files. If people want to backfill their data, they can refresh; connectors are meant to ingest data created after the connection is initialized.